mage_integrations/mage_integrations/sources/stripe/__init__.py

```diff
@@ -1121,20 +1136,71 @@ def write_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name
     # Write the parent bookmark value only when the parent is selected
     if not is_sub_stream:
         singer.write_bookmark(Context.state,
-                              stream_name + '_events',
+                              stream_name,
                               'updates_created',
                               max_created)
 
     # Write the child bookmark value only when the child is selected
     if sub_stream_name and Context.is_selected(sub_stream_name):
         singer.write_bookmark(Context.state,
-                              sub_stream_name + '_events',
+                              sub_stream_name,
                               'updates_created',
                               max_created)
 
     singer.write_state(Context.state)
 
 
+def reset_bookmark_for_event_updates(is_sub_stream, stream_name, sub_stream_name, start_date):
+    """
+    Reset bookmark for parent and child streams
+    to start date and clear the bookmark date for event updates.
+    """
+    # Write the parent bookmark value only when the parent is selected
+    if not is_sub_stream:
+        singer.write_bookmark(Context.state,
+                              stream_name,
+                              STREAM_REPLICATION_KEY.get(stream_name),
+                              start_date)
+        Context.state.get("bookmarks").pop(stream_name, None)
+
+    # Write the child bookmark value only when the child is selected
+    if sub_stream_name and Context.is_selected(sub_stream_name):
+        singer.write_bookmark(Context.state,
+                              sub_stream_name,
+                              STREAM_REPLICATION_KEY.get(sub_stream_name),
+                              start_date)
+        Context.state.get("bookmarks").pop(sub_stream_name, None)
+
+    singer.write_state(Context.state)
+
+
+def sync(logger=LOGGER):
```

Comment:
This method is not used here

Reply:
Removed on 028336556fca308242837d69d26186c17481362d